superset/db_engine_specs/base.py

```diff
@@ -403,6 +401,8 @@ def _allowed_file(filename: str) -> bool:
                 extension is not None and extension[1:] in config["ALLOWED_EXTENSIONS"]
             )
 
+        database = db.session.query(Database).filter_by(id=form.con.data.id).one()
```

Comment:
The caller has the database object handy, maybe it should just pass it over? The interface is a bit funky here (receiving the `form` object) but preventing from re-fetching the database object from the database is positive and prevents the need for the late import that works around the circular imports...

Reply:
Sure. Can do. I agree the interface is wonky here - dealing with the form object should be done at the view layer. I'll make a note to refactor the interface in a future PR.